Lib/platform.py

```diff
@@ -170,10 +170,13 @@ def libc_ver(executable=sys.executable, lib='', version='',
                 m = _libc_search.search(binary, pos)
             else:
                 m = None
-            if not m:
-                binary = f.read(chunksize)
-                if not binary:
-                    break
+            if not m or (not eof and m.end() == len(binary)):
+                chunk = f.read(chunksize)
+                if not chunk:
+                    if not m:
+                        break
+                    eof = True
+                binary = binary[max(pos, len(binary) - 1000):] + chunk
```

Comment:
Maybe it's better to put this line in an `else` block. No meaning to concatenate an empty bytes.